### PR TITLE
[MR][HMA-1585] QA Fixes

### PR DESCRIPTION
**Pull request by mrolfe88**
Added closed account check to BonusPeriod Milestone logic
Fixed date logic
Fixed date formatting
Updated/added unit tests

### CODE REVIEW

File: app/uk/gov/hmrc/mobilehelptosave/services/AccountService.scala [hunk 1]
```diff
@@ -90,7 +90,7 @@ class HtsAccountService[F[_]](
           case Some(account) =>
             EitherT.liftF[F, ErrorInfo, Option[Account]](for {
               _ <- if (milestonesConfig.balanceMilestoneCheckEnabled) milestonesService.balanceMilestoneCheck(nino, account.balance) else F.pure(())
-              _ <- if (milestonesConfig.bonusPeriodMilestoneCheckEnabled)
+              _ <- if (milestonesConfig.bonusPeriodMilestoneCheckEnabled && !account.isClosed)
```
Comment (peter-hazell) [inline app/uk/gov/hmrc/mobilehelptosave/services/AccountService.scala]: I think this needs to happen on the balance check as well

Comment (mrolfe88) [reply app/uk/gov/hmrc/mobilehelptosave/services/AccountService.scala]: Apparently not... Confirmed by Kam/Andy on the Jira ticket: https://jira.tools.tax.service.gov.uk/browse/HMA-1585

Comment (peter-hazell) [reply app/uk/gov/hmrc/mobilehelptosave/services/AccountService.scala]: Fair enough! Approved

File: app/uk/gov/hmrc/mobilehelptosave/services/MilestonesService.scala [hunk 2]
```diff
@@ -129,10 +129,10 @@ class HtsMilestonesService[F[_]](
...
     val hasSecondBonusEstimate             = secondPeriodBonusEstimate > 0
     val firstPeriodBonusPaid               = firstPeriodBonus > 0
-    val within20DaysOfFirstPeriodEndDate   = currentDateInDuration(endOfFirstBonusPeriod, 20)
-    val under90DaysSinceFirstPeriodEndDate = currentDateInDuration(endOfFirstBonusPeriod.plusDays(91), 90)
-    val within20DaysOfFinalEndDate         = currentDateInDuration(endOfSecondBonusPeriod, 20)
-    val dateFormat                         = DateTimeFormatter.ofPattern("d-MMM-yyyy")
+    val within20DaysOfFirstPeriodEndDate   = currentDateInDuration(endOfFirstBonusPeriod, 19)
+    val under90DaysSinceFirstPeriodEndDate = currentDateInDuration(firstPeriodBonusPaidOnOrAfterDate.plusDays(90), 89)
+    val within20DaysOfFinalEndDate         = currentDateInDuration(endOfSecondBonusPeriod, 19)
+    val dateFormat                         = DateTimeFormatter.ofPattern("d-MMMM-yyyy")
```
Comment (ngoulongkam) [inline app/uk/gov/hmrc/mobilehelptosave/services/MilestonesService.scala]:
```suggestion
    val dateFormat                         = DateTimeFormatter.ofPattern("d MMMM yyyy")
```

The suggestion Andy made on Slack doesn't have dashes in between the date.
https://hmrcdigital.slack.com/archives/G7L9K3C6S/p1566914756006600

File: test/uk/gov/hmrc/mobilehelptosave/services/MilestonesServiceSpec.scala [hunk 3]
```diff
@@ -189,6 +189,8 @@ class MilestonesServiceSpec
       val milestonesRepo      = fakeMilestonesRepo(List.empty)
       val previousBalanceRepo = fakePreviousBalanceRepo(Some(PreviousBalance(nino, 0, LocalDateTime.now())))
 
+      println("Date = " + baseBonusTerms(0).endDate)
```
Comment (ngoulongkam) [inline test/uk/gov/hmrc/mobilehelptosave/services/MilestonesServiceSpec.scala]: Is this println meant to be here?

File: it/uk/gov/hmrc/mobilehelptosave/MilestonesISpec.scala [hunk 4]
```diff
@@ -43,8 +44,9 @@ class MilestonesISpec
 
   override implicit lazy val app: Application = appBuilder.build()
 
-  private val generator = new Generator(Random.nextInt())
-  private val journeyId = randomUUID().toString
+  private val generator  = new Generator(Random.nextInt())
+  private val journeyId  = randomUUID().toString
+  private val dateFormat = DateTimeFormatter.ofPattern("d-MMMM-yyyy")
```
Comment (ngoulongkam) [inline it/uk/gov/hmrc/mobilehelptosave/MilestonesISpec.scala]:
```suggestion
  private val dateFormat = DateTimeFormatter.ofPattern("d MMMM yyyy")
```
Same for here.

**Pull request merged**